ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/SqlLedger.scala

```diff
@@ -62,43 +63,204 @@ object SqlLedger {
       eventsPageSize: Int,
       metrics: Metrics,
       lfValueTranslationCache: LfValueTranslation.Cache
-  )(implicit mat: Materializer, logCtx: LoggingContext): ResourceOwner[Ledger] =
-    for {
-      _ <- ResourceOwner.forFuture(() => new FlywayMigrations(jdbcUrl).migrate()(DEC))
-      ledgerDao <- JdbcLedgerDao.validatingWriteOwner(
-        serverRole,
-        jdbcUrl,
-        eventsPageSize,
-        metrics,
-        lfValueTranslationCache)
-      ledger <- ResourceOwner.forFutureCloseable(
-        () =>
-          new SqlLedgerFactory(ledgerDao).createSqlLedger(
-            ledgerId,
+  )(implicit mat: Materializer, logCtx: LoggingContext)
+      extends ResourceOwner[Ledger] {
+
+    private val logger = ContextualizedLogger.get(this.getClass)
+
+    override def acquire()(implicit executionContext: ExecutionContext): Resource[Ledger] =
+      for {
+        _ <- Resource.fromFuture(new FlywayMigrations(jdbcUrl).migrate())
+        ledgerDao <- JdbcLedgerDao
+          .validatingWriteOwner(
+            serverRole,
+            jdbcUrl,
+            eventsPageSize,
+            metrics,
+            lfValueTranslationCache,
+          )
+          .acquire()
+        _ <- startMode match {
+          case SqlStartMode.AlwaysReset =>
+            Resource.fromFuture(ledgerDao.reset())
+          case SqlStartMode.ContinueIfExists =>
+            Resource.unit
+        }
+        ledgerId <- Resource.fromFuture(
+          initialize(
+            initialLedgerId,
             participantId,
             timeProvider,
-            startMode,
             acs,
             packages,
             initialLedgerEntries,
-            queueDepth,
-            transactionCommitter,
-        ))
-    } yield ledger
+            ledgerDao,
+          ))
+        ledgerEnd <- Resource.fromFuture(ledgerDao.lookupLedgerEnd())
+        ledgerConfig <- Resource.fromFuture(ledgerDao.lookupLedgerConfiguration())
+        dispatcher <- ResourceOwner
+          .forCloseable(() => Dispatcher[Offset]("sql-ledger", Offset.beforeBegin, ledgerEnd))
```

Comment:
Given that there's a string and two parameters with the same type, named parameters could make sense here.

Reply:
Yup! I added named parameters for all calls to the dispatcher constructor that I could find.